lib/streamlit/report_session.py

```diff
@@ -243,7 +243,7 @@ def session_state(self) -> "SessionState":
     def _on_source_file_changed(self):
         """One of our source files changed. Schedule a rerun if appropriate."""
         if self._run_on_save:
-            self.request_rerun()
+            self.request_rerun(self._client_state)
```

Comment:
Interesting.. do you think this was setup like this with forward looking intentions, or is it missing more likely an oversight?

Reply:
There are other calls to `request_rerun` in this file that pass a `client_state` argument, so that's why the method already takes an optional argument.

The client follows the convention that when it doesn't receive an updated `client_state` from the server, it just uses what it already has, and previously when you updated a widget's default argument, you were actually seeing a new widget when the script reran, so those two things combined allowed us to not have to worry about updating client widget state in the auto-rerun case.

Reply:
Oh weird, I only see one reference to that method when I searched :hmm: